src/HypothesisTests.jl

```diff
@@ -72,32 +72,38 @@ end
 function Base.show{T<:HypothesisTest}(io::IO, test::T)
     println(io, testname(test))
     println(io, repeat("-", length(testname(test))))
+    
+    # utilities for pretty-printing
+    conf_string = string(floor((1 - alpha(test)) * 100, 6)) # limit to 6 decimals in %
+    format_detail(label::String, value::Any, len::Int) = # len is max length of label
+        "    " * label * " "^max(len - length(label), 0) * string(value)
 
     # population details
-    has_ci = applicable(StatsBase.confint, test)
+    has_ci = applicable(confint, test)
     (param_name, param_under_h0, param_estimate) = population_param_of_interest(test)
     println(io, "Population details:")
-    println(io, "    parameter of interest:   $param_name")
-    println(io, "    value under h_0:         $param_under_h0")
-    println(io, "    point estimate:          $param_estimate")
+    println(io, format_detail("parameter of interest:", param_name, 32))
+    println(io, format_detail("value under h_0:", param_under_h0, 32))
+    println(io, format_detail("point estimate:", param_estimate, 32))
     if has_ci
-        println(io, "    95% confidence interval: $(StatsBase.confint(test))")
+        println(io, format_detail(conf_string*"% confidence interval:", confint(test), 32))
     end
     println(io)
 
     # test summary
-    p = pvalue(test)
-    outcome = if p > 0.05 "fail to reject" else "reject" end
-    tail = default_tail(test)
-    println(io, "Test summary:")
-    println(io, "    outcome with 95% confidence: $outcome h_0")
-    if tail == :both
-        println(io, "    two-sided p-value:           $p")
-    elseif tail == :left || tail == :right
-        println(io, "    one-sided p-value:           $p")
+    testtail = tail(test)
+    p = pvalue(test) # obeys value of tail(test) if applicable
+    outcome = p > alpha(test) ? "fail to reject" : "reject"
+    if testtail == :both
+        plabel ="two-sided p-value:"
+    elseif testtail == :left || testtail == :right
+        plabel = "one-sided p-value ($(string(testtail)) tail):"
     else
-        println(io, "    p-value:                     $p")
+        plabel = "p-value:"
     end
+    println(io, "Test summary:")
+    println(io, format_detail("outcome with "*conf_string*"% confidence:", outcome*" h_0", 36))
```

Comment:
Add spaces around `*` for consistency and readability. Here you can actually use string interpolation instead.